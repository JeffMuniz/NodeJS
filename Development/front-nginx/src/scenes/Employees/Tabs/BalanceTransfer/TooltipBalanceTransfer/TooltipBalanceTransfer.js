import React, { Fragment } from "react";
import { func, string } from "prop-types";
import { Button } from "@common";
import { buttonTypes as buttonTypesEnum } from "@enums";

import {
  Container,
  Text,
  Overlay,
  ButtonWrapper,
  TextWrapper,
} from "./TooltipBalanceTransfer.styles";

const TooltipBalanceTransfer = ({ close, typeTooltip }) => {
  const helpTransfer = () => (
    <Fragment>
      <Text>
        A transferência de saldo são vinculados ao CPF do portador e CNPJ da
        empresa.
      </Text>
      <Text>
        O RH pode determinar um valor mínimo que deve permanecer em cada
        maceficio para uso especifico dentro da rede de alimentação e/ou
        refeição, respeitando os acordos sindicais. Ex:
      </Text>
      <Text>Valor mínimo cadastrado pelo RH no Alimentação: R$200,00</Text>
      <Text>Saldo do cartão do colaborador Carlos: R$700,00</Text>
      <Text>Valor disponível para transferência para o Carlos: R$500,00</Text>
      <Text>
        O valor de R$200,00 ficará no cartão para que o Carlos utilize apenas na
        rede de alimentação.
      </Text>
      <Text>
        Se nenhum valor for cadastrado pelo RH, o portador poderá transferir o
        saldo total do seu maceficio para o outro produto.
      </Text>
      <Text>
        O RH poderá editar as regras voltando a esta tela para gerenciamento.
      </Text>
      <Text>
        Ao aceitar os termos e Salvar, as edições já estarão disponíveis para
        todos os seus colaboradores
      </Text>
    </Fragment>
  );

  const infoBalanceTransfer = () => (
    <Fragment>
      <Text>
        Para habilitar essa funcionalidade aos seus colaboradores você precisa
        ter os dois maceficios contratados:
      </Text>
      <Text>
        Alimentação e Refeição, desta forma seus colaboradores poderão fazer
        transferencia de saldo entre seus cartões.
      </Text>
    </Fragment>
  );

  return (
    <Fragment>
      <Container
        className={typeTooltip === "hasBalanceTransfer" ? "min-container" : ""}
      >
        <TextWrapper>
          {typeTooltip === "transferLimit" && helpTransfer()}
          {typeTooltip === "hasBalanceTransfer" && infoBalanceTransfer()}
        </TextWrapper>
        <ButtonWrapper>
          <Button
            id="button_submit"
            onPress={() => close()}
            value="Fechar"
            buttonType={buttonTypesEnum.primary}
          />
        </ButtonWrapper>
      </Container>
      <Overlay onClick={() => close()} />
    </Fragment>
  );
};

TooltipBalanceTransfer.propTypes = {
  close: func.isRequired,
  typeTooltip: string.isRequired,
};

export default TooltipBalanceTransfer;
