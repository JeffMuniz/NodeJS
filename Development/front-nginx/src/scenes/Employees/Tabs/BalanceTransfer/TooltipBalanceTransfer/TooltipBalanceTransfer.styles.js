import styled from "styled-components";
import { dustyGray } from "@colors";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  background: #fff;
  z-index: 5;
  box-shadow: 0 3px 6px rgb(0 0 0 / 16%), 0 3px 6px rgb(0 0 0 / 23%);
  border-radius: 10px;
  width: 568px;
  height: 408px;
  top: -135px;
  left: 225px;

  &::before {
    content: "";
    position: absolute;
    width: 0;
    height: 0;
    bottom: 100%;
    top: 167px;
    border: 0.75rem solid transparent;
    border-left: none;
    left: -12px;
    border-right-color: #fff;
    z-index: 5;
  }

  &.min-container {
    width: 466px;
    background: #fff;
    height: 175px;
    top: -82px;
    left: -4px;

    &::before {
      top: 73px;
    }
  }
`;

export const Text = styled.p`
  font-size: 13px;
  color: ${dustyGray};
  margin: 8px 5px;
`;

export const Overlay = styled.div`
  display: block;
  position: fixed;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.6);
  z-index: 4;
  top: 0;
  left: 0;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  width: 95%;
  margin: 0;
`;

export const TextWrapper = styled.div`
  padding: 10px;
`;
