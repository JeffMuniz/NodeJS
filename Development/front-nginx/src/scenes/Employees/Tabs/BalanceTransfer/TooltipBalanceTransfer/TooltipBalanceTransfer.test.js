import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import TooltipBalanceTransfer from "./TooltipBalanceTransfer";

describe("Tooltip Balance Transfer - Unit Tests", () => {
  it("Should render Tooltip Balance Transfer", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<TooltipBalanceTransfer close={() => jest.fn()} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
