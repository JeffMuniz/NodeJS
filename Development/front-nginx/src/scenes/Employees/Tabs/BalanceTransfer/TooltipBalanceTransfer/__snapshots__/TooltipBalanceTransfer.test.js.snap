// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Balance Transfer - Unit Tests Should render Tooltip Balance Transfer 1`] = `
<UNDEFINED>
  <styled.div
    className=""
  >
    <styled.div />
    <styled.div>
      <Button
        action={null}
        buttonType="primary"
        disabled={false}
        id="button_submit"
        imgColor={null}
        imgSrc={undefined}
        imgWidth={16}
        loading={false}
        onPress={[Function]}
        popupText={null}
        value="Fechar"
      />
    </styled.div>
  </styled.div>
  <styled.div
    onClick={[Function]}
  />
</UNDEFINED>
`;
